### ISSUE TITLE
Add ability to control how exceptions are grouped for error_grouping

### ISSUE THREAD

**Issue by westonganger** [smartinez87/exception_notification#513]
We should be able to control how the exceptions are being grouped. 

Out of the box the errors are grouped by name and message. This is not very unique.

I want to be able to provide a proc to generate custom grouping identifiers

```ruby
get_group_identifier: ->(exception, env, data){
  [
    exception.name,
    exception.message,
    env.ip,
    exception.backtrace.first,
  ].join("/")
}
```

Without this feature the error grouping feature seems not very useful for generic use.

**Comment by OskarEichler** [comment]
Yes this would be great - we enabled error_grouping and it doesn't seem to do anything. Getting 1000s of emails of the same errors within minutes -.-

@Martin91 I know that you originally implemented this - would it be possible to add options for more specific grouping?
